pontoon/machinery/views.py

```diff
@@ -113,6 +117,30 @@ def machine_translation(request):
         return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
 
 
+def machine_translation_caighdean(request):
+    """Get translation from Caighdean machine translation service."""
+    try:
+        entityid = int(request.GET['id'])
+    except (MultiValueDictKeyError, ValueError) as e:
+        return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
+
+    try:
+        entity = Entity.objects.get(id=entityid)
+    except Entity.DoesNotExist as e:
+        return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
+
+    try:
+        text = entity.translation_set.get(locale__code='gd').string
+    except Translation.DoesNotExist:
+        return JsonResponse({})
+
+    try:
+        translation = Translator().translate(text)
+        return JsonResponse({'original': text, 'translation': translation})
+    except TranslationError as e:
+        return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
```

Comment:
Yeah, it is weird to return a BadRequest here. It's not the user's fault if there are not translations available for that input.

Reply:
yep